### PR TITLE
crystal-lang: vendor libevent 2.0.22 during the build

### PR DESCRIPTION
**Pull request by ilovezfs**
The crystal-lang bootstrap binary is linked against our libevent 2.0.22,
so this will unblock the libevent 2.1.8 PR.

### CODE REVIEW

**Comment by ilovezfs** [comment]
@asterite the crystal-lang bootstrap binary is causing :boom: in https://github.com/Homebrew/homebrew-core/pull/9344 because it is dynamically linked against Homebrew's libevent 2.0.22 (libevent-2.0.5.dylib). This PR is one way to work around the situation. Alternatively, we could wait for a new bootstrap binary from you that is linked against libevent statically.

Either way, it would be great if the next iteration of your bootstrap binary were statically linked against libevent so that this doesn't happen again. I presume we have the same problem for libgc.1.dylib (bdw-gc formula) the next time that gets an ABI breaking upgrade.

**Comment by asterite** [comment]
@ilovezfs Hm, that's going to be tough, because there's no easy way to say "I want to statically link this library" in Crystal. I guess it's time we start thinking how to do that...

File: Formula/crystal-lang.rb [hunk 1]
```diff
@@ -33,9 +33,29 @@ class CrystalLang < Formula
...
+        system "make"
+        system "make", "install"
+      end
+    end
+
     (buildpath/"boot").install resource("boot")
 
+    macho = MachO.open("#{buildpath}/boot/embedded/bin/crystal")
+    macho.change_dylib("/usr/local/opt/libevent/lib/libevent-2.0.5.dylib",
+                       "#{buildpath}/vendor/libevent/lib/libevent-2.0.5.dylib")
```
Comment (nijikon) [inline Formula/crystal-lang.rb]: Why there is `2.0.5` here if we build agains `2.0.22`?

Comment (nijikon) [inline Formula/crystal-lang.rb]: Nevermind, this is weird.

```
bash-3.2$ ls libevent*.dylib
libevent-2.0.5.dylib		libevent_core-2.0.5.dylib	libevent_extra-2.0.5.dylib	libevent_openssl-2.0.5.dylib	libevent_pthreads-2.0.5.dylib
libevent.dylib			libevent_core.dylib		libevent_extra.dylib		libevent_openssl.dylib		libevent_pthreads.dylib
```

**Pull request closed without merging**